framework/src/org/checkerframework/common/value/util/Range.java

```diff
@@ -767,6 +767,38 @@ public Range refineEqualTo(Range right) {
     }
 
     /**
+     * Refines this range to reflect that no value in it can be equal to a value in the given range.
+     * This is used for calculating the control-flow-refined result of the != operator. For example:
+     *
+     * <pre>
+     * <code>
+     *     {@literal @}IntRange(from = 0, to = 10) int a;
+     *     {@literal @}IntRange(from = 3, to = 15) int b;
+     *     ...
+     *     if (a != b) {
+     *         // range of <i>a</i> is now refined to [0, 2] because a value in range [3, 10]
```

Comment:
I don't think this is the correct logic. `a` should not be refined to `[0, 2]` because even if `b` has range of `[3, 15]` and `a` is not equal to `b`, `a` still has a chance to equal to `3` (or other values) because `b` is not guaranteed to be `3`. 

This is the reason why I didn't implement this api. An equality check can't really provide much information for range refinement. Do you agree?